packages/core/src/build-stage.ts

```diff
@@ -35,7 +35,7 @@ export default class BuildStage<NamedTrees> implements Stage {
         this.packageCache = packageCache;
         this.active = await this.instantiate(outputPath, this.prevStage.inputPath, packageCache);
       }
-      delete treePaths.__prevStageTree;
+      delete (treePaths as any).__prevStageTree;
```

Comment:
Here we are intentionally changing a thing of one type into another type.

The types were both correct, we don't want `__prevStageTree` to be optional. Instead, we need to escape the type system just during the middle of the conversion between types. Everybody outside these functions will only see consistent types.

Reply:
:+1: